### PR TITLE
Head idempotent fixes: salt_states_catalog.feature idempotent + fixes

### PR DESCRIPTION
**Pull request by meaksh**

### CODE REVIEW

File: features/step_definitions/salt_steps.rb [hunk 1]
```diff
@@ -854,12 +843,24 @@
 
 SALT_PACKAGES = "salt salt-minion".freeze
 
-Given(/^no Salt packages are installed on remote "(.*)"$/) do |host|
-  if host == "ssh-minion"
-    $ssh_minion.run("test -e /usr/bin/zypper && zypper --non-interactive remove -y #{SALT_PACKAGES}", false)
-  end
-  if host == "centos"
-    $ceos_minion.run("test -e /usr/bin/yum && yum -y remove #{SALT_PACKAGES}", false)
+Given(/^no Salt packages are installed on "(.*)"$/) do |host|
```
Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: use this here for refractoring this function

https://github.com/SUSE/spacewalk-testsuite-base/blob/d3392f6518166c10aafdb4560b6861c268d4cecb/features/step_definitions/command_steps.rb#L14

File: run_sets/testsuite.yml [hunk 2]
```diff
@@ -43,6 +43,8 @@
 - features/salt.feature
 - features/salt_minions_page.feature
 - features/salt_minion_details.feature
+- features/salt_download_endpoint.feature
```
Comment (MalloZup) [inline run_sets/testsuite.yml]: @meaksh  can you put the feature salt randomly ordered in idempotent? so we can assume that they are idempotent and not order dipendent 🎏

Comment (meaksh) [inline run_sets/testsuite.yml]: Exactly. You can change the execution order if you want, they would pass as they're now idempotent 😎

**Comment by MalloZup** [comment]
@meaksh ok looks good, keep moving ! 👍 

cel

**Pull request merged**